temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java

```diff
@@ -278,9 +279,20 @@ public String getTaskQueue() {
     return null;
   }
 
+  /** Returns name of the task queue that test worker is polling. */
+  public String getTaskQueue() {
+    return taskQueue;
+  }
+
   /** Returns client to the Temporal service used to start and query workflows. */
   public WorkflowClient getWorkflowClient() {
-    return testEnvironment.getWorkflowClient();
+    if (useExternalService) {
```

Comment:
Why do we need this? TestWorkflowEnvironment already handles workflow stubs creation and WorkflowClient creation.

Reply:
I was trying to move this functionality from the individual tests that checked useExternalService (eg testChildWorkflowExecutionPromiseHandler()) into the rule and keep the code as readable as possible. Are you saying testEnvironment.getWorkflowClient() does exactly the same thing as in the other part of this if statement? If so you may be right and we don't need it.

Reply:
Please take a look at [this code](https://github.com/temporalio/sdk-java/blob/ddeaa5f23eda9113fa23a32e924c078f1a38ae0c/temporal-testing/src/main/java/io/temporal/testing/TestWorkflowEnvironmentInternal.java#L75)